apps/hash-frontend/src/components/hooks/block-protocol-functions/knowledge/knowledge-shim.ts

```diff
@@ -6,38 +6,55 @@
  * package and be removed from here.
  */
 
-import { MessageCallback } from "@blockprotocol/core";
+import { MessageCallback, MessageReturn } from "@blockprotocol/core";
 import {
   CreateResourceError,
   ReadOrModifyResourceError,
 } from "@blockprotocol/graph";
+import { VersionedUri } from "@blockprotocol/type-system";
 import {
   Entity,
   EntityId,
+  EntityPropertiesObject,
+  EntityRootType,
   LinkData,
-  PropertyObject,
   Subgraph,
-  SubgraphRootTypes,
-  VersionedUri,
-} from "@local/hash-subgraph/main";
+} from "@local/hash-subgraph";
 
-export type KnowledgeCallbacks = {
-  getEntity: GetEntityMessageCallback;
-  createEntity: CreateEntityMessageCallback;
-  aggregateEntities: AggregateEntitiesMessageCallback;
-  archiveEntity: ArchiveEntityMessageCallback;
-};
+/* Entity CRU */
 
-export type GetEntityRequest = {
+export type GetEntityData = {
   entityId: EntityId;
   graphResolveDepths?: Partial<Subgraph["depths"]>;
 };
 
-/* Entity CRU */
+/** @todo-0.3 - we really want some type safety on these if we can find it, but this doesn't work */
+// export type GetEntityMessageCallback = Subtype<
+//   EmbedderGraphMessageCallbacks<true>["getEntity"],
+//   MessageCallback<
+//     GetEntityRequest,
+//     null,
+//     MessageReturn<Subgraph<EntityRootType>>,
+//     ReadOrModifyResourceError
+//   >
+// >;
```

Comment:
This feels _**very**_ unfortunate and somewhat important. We can't make sure that our custom callbacks are compatible with the ones from the BP because our local `Subgraph` isn't assignable to the `Subgraph` from the BP.

Maybe make a custom `Subtype` helper specifically for these which ignores `Subgraph` or something?

This is important if we continue to enhance the typings of the BP hooks rather than using the vanilla ones (which is somewhat needed if we keep using the hooks across the frontend rather than just at the EA/block boundary)